### ISSUE TITLE
running tests on mac

### ISSUE THREAD

**Issue by offerm** [raiden-network/raiden#4722]
I'm using mac for Raiden development.
geth version is 1.8.0

Trying to run `raiden/tests/integration/api/test_restapi.py::test_api_payments_with_hash_no_secret`

getting assert error"
```

raiden/tests/integration/fixtures/blockchain.py:91: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
/usr/local/Cellar/python/3.7.3/Frameworks/Python.framework/Versions/3.7/lib/python3.7/contextlib.py:112: in __enter__
    return next(self.gen)
raiden/tests/utils/eth_node.py:566: in run_private_blockchain
    with runner as executors:
/usr/local/Cellar/python/3.7.3/Frameworks/Python.framework/Versions/3.7/lib/python3.7/contextlib.py:112: in __enter__
    return next(self.gen)
raiden/tests/utils/eth_node.py:429: in eth_run_nodes
    nodes_configuration, eth_node_descs, base_datadir, genesis_file, chain_id, verbosity
raiden/tests/utils/eth_node.py:390: in eth_nodes_to_cmds
    geth_prepare_datadir(datadir, genesis_file)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

datadir = '/private/var/folders/06/5x3r2f911slcdg34vqnt058h0000gn/T/pyt-admin/36/test_api_paymen0/7524840c'
genesis_file = '/private/var/folders/06/5x3r2f911slcdg34vqnt058h0000gn/T/pyt-admin/36/test_api_paymen0/custom_genesis.json'

    def geth_prepare_datadir(datadir: str, genesis_file: str) -> None:
        node_genesis_path = os.path.join(datadir, "custom_genesis.json")
        ipc_path = datadir + "/geth.ipc"
>       assert len(ipc_path) < 104, f'geth data path "{ipc_path}" is too large'
E       AssertionError: geth data path "/private/var/folders/06/5x3r2f911slcdg34vqnt058h0000gn/T/pyt-admin/36/test_api_paymen0/7524840c/geth.ipc" is too large
E       assert 104 < 104
E        +  where 104 = len('/private/var/folders/06/5x3r2f911slcdg34vqnt058h0000gn/T/pyt-admin/36/test_api_paymen0/7524840c/geth.ipc')

raiden/tests/utils/eth_node.py:370: AssertionError
=========================== 1 error in 1.90 seconds ============================
Process finished with exit code 0
```

If I comment out this assert I'm getting this:

```

raiden/tests/integration/fixtures/blockchain.py:91: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
/usr/local/Cellar/python/3.7.3/Frameworks/Python.framework/Versions/3.7/lib/python3.7/contextlib.py:112: in __enter__
    return next(self.gen)
raiden/tests/utils/eth_node.py:566: in run_private_blockchain
    with runner as executors:
/usr/local/Cellar/python/3.7.3/Frameworks/Python.framework/Versions/3.7/lib/python3.7/contextlib.py:112: in __enter__
    return next(self.gen)
raiden/tests/utils/eth_node.py:452: in eth_run_nodes
    stack.enter_context(executor)
/usr/local/Cellar/python/3.7.3/Frameworks/Python.framework/Versions/3.7/lib/python3.7/contextlib.py:426: in enter_context
    result = _cm_type.__enter__(cm)
../.virtualenvs/raiden/lib/python3.7/site-packages/mirakuru/base.py:136: in __enter__
    return self.start()
raiden/utils/http.py:115: in start
    self.wait_for(self.check_subprocess)
raiden/utils/http.py:156: in wait_for
    if wait_for():
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <raiden.utils.http.JSONRPCExecutor: "geth --nod..." 0x10cbd3ba8>

    def check_subprocess(self):
        """
        Make sure the process didn't exit with an error and run the checks.
    
        :rtype: bool
        :return: the actual check status
        :raise ProcessExitedWithError: when the main process exits with
            an error
        """
        exit_code = self.process.poll()
        if exit_code is not None and exit_code != 0:
            # The main process exited with an error. Clean up the children
            # if any.
            self._kill_all_kids(self._sig_kill)
            self._clear_process()
>           raise ProcessExitedWithError(self, exit_code)
E           mirakuru.exceptions.ProcessExitedWithError: The process invoked by the <raiden.utils.http.JSONRPCExecutor: "geth --nodekeyhex 7524840c50b30aed64bf447104f7597252d4941ba6a134dc021844bc949d5bd4 --port 8501 --rpcport 8500 --bootnodes enode://67e3448c383e3f074df6ebd8d7fe67b6ded459b27ed91da10061f7837aa8c5a7febcbbbc5466bbaf44eca0b2a913ee7365900dd575803061b98e931bb1c6e4ea@127.0.0.1:8501 --unlock 0x866bAB5D7e0fBf89Eb2AF67A28c2248398197256 --password /private/var/folders/06/5x3r2f911slcdg34vqnt058h0000gn/T/pyt-admin/37/test_api_paymen0/pw --nodiscover --rpc --rpcapi eth,net,web3,personal,txpool --rpcaddr 127.0.0.1 --networkid 337 --verbosity 3 --datadir /private/var/folders/06/5x3r2f911slcdg34vqnt058h0000gn/T/pyt-admin/37/test_api_paymen0/7524840c --mine"> executor has exited with a non-zero code: 1.

../.virtualenvs/raiden/lib/python3.7/site-packages/mirakuru/base.py:449: ProcessExitedWithError
=========================== 1 error in 4.76 seconds ============================
Process finished with exit code 0
```

in geth log file I see this:
```
INFO [09-03|08:31:34.465] Maximum peer count                       ETH=25 LES=0 total=25
INFO [09-03|08:31:34.475] Starting peer-to-peer node               instance=Geth/v1.8.20-stable/darwin-amd64/go1.11.2
INFO [09-03|08:31:34.475] Allocated cache and file handles         database=/private/var/folders/06/5x3r2f911slcdg34vqnt058h0000gn/T/pyt-admin/38/test_api_paymen0/7524840c/geth/chaindata cache=512 handles=4611686018427387903
INFO [09-03|08:31:34.486] Initialised chain configuration          config="{ChainID: 337 Homestead: 0 DAO: <nil> DAOSupport: false EIP150: 0 EIP155: 0 EIP158: 0 Byzantium: 0 Constantinople: 0 Engine: clique}"
INFO [09-03|08:31:34.486] Initialising Ethereum protocol           versions="[63 62]" network=337
INFO [09-03|08:31:34.540] Loaded most recent local header          number=0 hash=487c98…db8e53 td=1 age=50y4mo3w
INFO [09-03|08:31:34.540] Loaded most recent local full block      number=0 hash=487c98…db8e53 td=1 age=50y4mo3w
INFO [09-03|08:31:34.540] Loaded most recent local fast block      number=0 hash=487c98…db8e53 td=1 age=50y4mo3w
INFO [09-03|08:31:34.541] Regenerated local transaction journal    transactions=0 accounts=0
INFO [09-03|08:31:34.541] Stored checkpoint snapshot to disk       number=0 hash=487c98…db8e53
INFO [09-03|08:31:34.546] New local node record                    seq=1 id=4ba0506f0690eb6f ip=127.0.0.1 udp=0 tcp=8501
INFO [09-03|08:31:34.547] Started P2P networking                   self="enode://67e3448c383e3f074df6ebd8d7fe67b6ded459b27ed91da10061f7837aa8c5a7febcbbbc5466bbaf44eca0b2a913ee7365900dd575803061b98e931bb1c6e4ea@127.0.0.1:8501?discport=0"
INFO [09-03|08:31:34.547] Blockchain manager stopped
INFO [09-03|08:31:34.547] Stopping Ethereum protocol
INFO [09-03|08:31:34.547] Ethereum protocol stopped
INFO [09-03|08:31:34.547] Transaction pool stopped
INFO [09-03|08:31:34.548] Database closed                          database=/private/var/folders/06/5x3r2f911slcdg34vqnt058h0000gn/T/pyt-admin/38/test_api_paymen0/7524840c/geth/chaindata
INFO [09-03|08:31:37.004] Mapped network port                      proto=tcp extport=8501 intport=8501 interface="UPNP IGDv1-IP1"
Fatal: Error starting protocol stack: listen unix /private/var/folders/06/5x3r2f911slcdg34vqnt058h0000gn/T/pyt-admin/38/test_api_paymen0/7524840c/geth.ipc: bind: invalid argument
```

please let me know what I'm doing wrong here.

Thanks

**Comment by hackaugusto** [comment]
@offerm This happens because MacOS's temp directory is quite large to begin with, `/private/var/folders/06/5x3r2f911slcdg34vqnt058h0000gn/T/` in you case. Because of that the `geth data path "/private/var/folders/06/5x3r2f911slcdg34vqnt058h0000gn/T/pyt-admin/36/test_api_paymen0/7524840c/geth.ipc" is too large`, and the ipc socket cannot be created (I believe the go runtime does that for compatibility reasons).

What you can do is set the tmp folder by hand, and run `pytest --basetemp=/tmp/raiden`:

```
  --basetemp=dir        base temporary directory for this test run.(warning:
                        this directory is removed if it exists)
```

**Comment by ulope** [comment]
Issue to track a proper fix: #4727.

I guess we can close this one.

**Comment by offerm** [comment]
Yep. --basetemp solves it.

Many thanks!
